tauri-app/src/lib/stores/settings.ts

```diff
@@ -4,59 +4,97 @@ import  find from 'lodash/find';
+    console.log(e);
+    return emptyConfig;
+  }
+}, { initial: emptyConfig });
 
-// chain
-export const activeChainSettingsIndex = cachedWritableInt("settings.activeChainIndex", 0);
-export const activeChainSettings = derived([settings, activeChainSettingsIndex], ([$settingsData, $activeChainSettingsIndex]) => $settingsData?.chains[$activeChainSettingsIndex]);
-export const rpcUrl = derived(activeChainSettings, ($activeChainSettings) => $activeChainSettings?.rpc_url);
-export const chainId = derived(activeChainSettings, ($activeChainSettings) => $activeChainSettings?.chain_id);
+// networks
+export const networks = derived(settings, ($settingsData) => Object.entries($settingsData?.networks));
+export const activeNetworkIndex = cachedWritableInt("settings.activeNetworkIndex", 0);
+export const activeNetwork = derived([networks, activeNetworkIndex], ([$networks, $activeNetworkIndex]) => $networks?.[$activeNetworkIndex]);
+export const rpcUrl = derived(activeNetwork, ($activeNetwork) => $activeNetwork?.[1].rpc);
+export const chainId = derived(activeNetwork, ($activeNetwork) => $activeNetwork?.[1].chain_id);
 export const activeChain = derived(chainId, ($activeChainId) => find(Object.values(chains), (c) => c.id === $activeChainId));
 export const activeChainHasBlockExplorer = derived(activeChain, ($activeChain) => {
-  return $activeChain && $activeChain.blockExplorers?.default !== undefined;
+  return $activeChain && $activeChain?.blockExplorers?.default !== undefined;
 });
-export const activeChainLatestBlockNumber = derived(activeChainSettings, ($activeChainSettings) => getBlockNumberFromRpc($activeChainSettings.rpc_url));
+export const activeChainLatestBlockNumber = derived(activeNetwork, ($activeNetwork) => getBlockNumberFromRpc($activeNetwork?.[1].rpc));
 
 // orderbook
-export const activeOrderbookSettingsIndex = cachedWritableInt("settings.activeOrderbookIndex", 0);
-export const activeOrderbookSettings =  derived([activeChainSettings, activeOrderbookSettingsIndex], ([$activeChainSettings, $activeOrderbookSettingsIndex]) => $activeChainSettings?.orderbooks[$activeOrderbookSettingsIndex]);
-export const subgraphUrl = derived(activeOrderbookSettings, ($activeOrderbookSettings) => $activeOrderbookSettings?.subgraph_url);
-export const orderbookAddress = derived(activeOrderbookSettings, ($activeOrderbookSettings) => $activeOrderbookSettings?.address);
+export const orderbooks = derived([settings, activeNetwork], ([$settingsData, $activeNetwork]) => Object.entries($settingsData.orderbooks).filter(v =>
+  // filter orderbooks based on active netowkr
+  v[1].network.rpc === $activeNetwork?.[1].rpc
+  && v[1].network.chain_id === $activeNetwork?.[1].chain_id
+  && v[1].network.label === $activeNetwork?.[1].label
+  && v[1].network.network_id === $activeNetwork?.[1].network_id
+  && v[1].network.currency === $activeNetwork?.[1].currency
+));
+export const activeOrderbookIndex = cachedWritableInt("settings.activeOrderbookIndex", 0);
```

Comment:
same note as above -- if we make this the string key rather than an array index it makes all the derived stuff so much cleaner :)

Reply:
yup